app/code/local/Mowdirect/Emailimporter/Model/Observer.php

```diff
@@ -5,48 +5,56 @@ class Mowdirect_Emailimporter_Model_Observer {
 
     public function handleEmailCron() {
         Mage::log('Stock importer (info):  cron started ');
-        $vendors = Mage::helper('emailimporter')->get_vendors();
+        $vendors = Mage::helper('emailimporter/Vendor')->get_vendors();
+        $vendor_model = Mage::getModel('udropship/vendor');
 
-        foreach ($vendors as $vendor) {
+        $vendor_to_update = new Varien_Data_Collection();
```

Comment:
@sudarshann, this is a generic collection so for code readability it should be `$vendors_to_update` i.e. the plural of vendors